### PR TITLE
Make pygraphviz really optional 

### PR DESCRIPTION
**Pull request by stellasia**
# Description

Pygraphviz is an optional dependency but was still imported in `pipeline.py`. This PR fixes that.

## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] Breaking change
- [ ] Documentation update
- [ ] Project configuration change

## Complexity

> **Note**
>
> Please provide an estimated complexity of this PR of either Low, Medium or High
>
>

Complexity: Low

## How Has This Been Tested?
- [x] Unit tests
- [ ] E2E tests
- [ ] Manual tests

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [ ] Unit tests have been updated
- [ ] E2E tests have been updated
- [ ] Examples have been updated
- [ ] New files have copyright header
- [ ] CLA (https://neo4j.com/developer/cla/) has been signed
- [ ] CHANGELOG.md updated if appropriate

### CODE REVIEW

File: src/neo4j_graphrag/experimental/pipeline/pipeline.py [hunk 1]
```diff
@@ -386,6 +390,12 @@ def draw(
         G.draw(path)
 
     def get_pygraphviz_graph(self, hide_unused_outputs: bool = True) -> pgv.AGraph:
+        if pgv is None:
+            raise ImportError(
+                "Could not import pygraphviz. "
+                "Follow installation instruction in pygraphviz documentation "
```
Comment (willtai) [inline src/neo4j_graphrag/experimental/pipeline/pipeline.py:396]: Could be useful to add a link here

Comment (stellasia) [reply src/neo4j_graphrag/experimental/pipeline/pipeline.py:396]: Not a big fan of adding links, it's already hard for us to update links to our own domain without adding the pain to maintain links to outside of our org :)

Comment (willtai) [reply src/neo4j_graphrag/experimental/pipeline/pipeline.py:396]: Perhaps a stackoverflow thread that discusses this? It's unlikely that will become outdated.

Comment (willtai) [reply src/neo4j_graphrag/experimental/pipeline/pipeline.py:396]: Imo anything to smoothen the user experience is useful here given encountering this installation issue is common

Comment (stellasia) [reply src/neo4j_graphrag/experimental/pipeline/pipeline.py:396]: Well, I also think developers should be able to google such things :) But added in the README [here](https://github.com/neo4j/neo4j-graphrag-python/pull/137/commits/b2c0f971eb9d4f6b85a5679cbbbf35d09aebdcff)

**Review by willtai — APPROVED** [review]

**Pull request merged**